sprint0/sprint0/Collision/LinkBorderCollisionHandler.cs

```diff
@@ -15,8 +15,11 @@ public void HandleCollision(IPlayer link, ISprite border, Direction side)
             }
             else if (border is LockedDoor lockedDoor)
             {
-                //TODO check if link has a key or the master key, if so, open door (decrement key if he doesn't have master key)
-                lockedDoor.OpenDoor();
+                if (link.HasKey() || link.HasItem(PlayerItems.MagicalKey))
+                {
+                    lockedDoor.OpenDoor();
+                    link.DecrementKey();
```

Comment:
doesn't this decrement key even when link uses the Magical key? and it doesn't decrement Magical key